.devops/nix/python-scripts.nix

```diff
@@ -0,0 +1,66 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  poetry-core,
+  mkShell,
+  python3Packages,
+  gguf-py,
+}@inputs:
+
+let
+  llama-python-deps = with python3Packages; [
+    numpy
+    sentencepiece
+    transformers
+    protobuf
+    torchWithoutCuda
+    gguf-py
+    tqdm
```

Comment:
I see `tiktoken` has been removed from the Python dependencies. It was **not** used by `gguf` or `llama.cpp` directly, but by some models during conversion for their tokenizer (like initially StableLM2 1.6B used `tiktoken`, although they later completely switched to `tokenizer.json` so technically `tiktoken` is no longer necessary for that model in particular).

See <https://github.com/ggerganov/llama.cpp/pull/5052#issuecomment-1902698030> for some background on this.

To be clear, I'm also okay with not including `tiktoken`, because It has been a while since I last saw a model which used this. It was added for convenience, but maybe it's not necessary.

Reply:
Good spot! I think its a good idea to keep tiktoken in the extra environments if it may be required by a script so I've added it as an additional input to those devShells in [e61d467](https://github.com/ggerganov/llama.cpp/pull/5664/commits/e61d4672dcb5f42fa9140e4b6d1ce6a20d8dac63)